agent/src/util/discovery_operator.rs

```diff
@@ -218,6 +222,25 @@ impl DiscoveryOperator {
                 result = stream.get_message() => {
                     let response = result?.ok_or_else(|| anyhow::anyhow!("Received response type None. Should not happen."))?;
                     trace!("internal_do_discover - got discovery results {:?}", response.devices);
+
+                    //if device external information update is detected, simply remove the old akri instance
```

Comment:
Why not update the instance instead? If we remove it, then it will also bring down associated broekrs

Reply:
I did not know we can update instance. Let me know more in the community meeting.